### PR TITLE
register enable custom function with a callback

### PR DESCRIPTION
**Pull request by pengwa**
**Description**: register enable custom function with a callback

1. fix regressions from: https://github.com/microsoft/onnxruntime/commit/419834d2859038165986567568a3bcc3e8c81681

 Previously, "enable_custom_autograd_support()" is enabled in ortmodule/__init__.py, registering the PythonOp custom exporter. 
 While the above change move this function calls into ORTModule creation, at which point, _enable_custom_autograd_function  is always is False (can be true, after ORTModule creation completed, and assign true to it's property). 

2. Enable auto grad Functions fallback test cases in CIs. 

Replacement of https://github.com/microsoft/onnxruntime/pull/8768

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py [hunk 1]
```diff
@@ -562,7 +565,8 @@ def input_generator():
     # Test pure inferencing scenarios, when inputs don't requires_grad.
     run_evaluate_test_and_compare(model_builder, input_generator, label_input)
 
-
+@pytest.mark.skipif(torch_version_lower_than("1.10.0"),
```
Comment (wschin) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py:568]: 1.19 or 1.20? We need to use the next version #.

Comment (pengwa) [reply orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py:568]: I think the next version will be 1.10

File: orttraining/orttraining/python/training/ortmodule/_custom_autograd_function.py [hunk 2]
```diff
@@ -18,3 +18,23 @@ def enable_custom_autograd_support():
...
+        return self._is_enabled
+
+    @enable_state.setter
+    def enable_state(self, new_val):
+        """Once state updated to True, autograd function runner and custom exporters are registered through callback.
+           This implies that enabling MUST be done before the ONNX model export.
+        """
+        self._is_enabled = new_val
+        if self._is_enabled:
+            self._callback()
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/training/ortmodule/_custom_autograd_function.py:40]:
```suggestion
            self._callback()

```
add new line

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py (hunk 1 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py:568]: yes, 1.10 will be the next pt version

File: orttraining/orttraining/test/python/_test_helpers.py [hunk 3]
```diff
@@ -194,10 +195,11 @@ def assert_values_are_close(input, other, rtol=1e-05, atol=1e-06):
         assert False, err_msg
 
 def enable_custom_autograd_function(module):
-    for mode in [True, False]:
-        module._torch_module._execution_manager(mode)._enable_custom_autograd_function = True
-        module._torch_module._execution_manager(mode)._save_onnx = True
-        module._torch_module._execution_manager(mode)._save_onnx_prefix = "utbench"
+    # load from json.
+    path_to_json = os.path.join(os.getcwd(), 'orttraining_test_ortmodule_autograd_config.json')
+    load_from_json(module, path_to_json)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/_test_helpers.py:200]: load from json is an experimental feature which can be deprecated. Relying on experimental code for a production feature is not adviced

**Review by thiagocrepaldi — DISMISSED** [review]

File: orttraining/orttraining/test/python/_test_helpers.py (hunk 3 above)
Comment (pengwa) [reply orttraining/orttraining/test/python/_test_helpers.py:200]: Then how should we enable the feature without load from json?

**Pull request closed without merging**